app/views/finance/invoices/show.html.haml

```diff
@@ -35,6 +35,7 @@
   %b= heading_helper(Invoice, :note) + ':'
   =h @invoice.note
 
-= link_to t('ui.edit'), edit_finance_invoice_path(@invoice)
-|
+- if current_user.role_finance? or (current_user.role_invoices? and not @invoice.paid_on)
```

Comment:
That's what [helpers](http://api.rubyonrails.org/classes/ActionController/Helpers.html) can be used for: app/helpers/finance/invoice_helper.rb in this case.